rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/kustomize/KustomizeTemplateUtils.java

```diff
@@ -0,0 +1,140 @@
+        try {
+            Path path = Paths.get(FilenameUtils.getPath(artifact.getReference()));
+            artifact.setReference(path.toString());
+            HashSet<String> files = getFilesFromArtifact(artifact);
+            artifacts = files.stream()
+              .map(f -> {
+                return Artifact.builder()
+                    .reference(f)
+                    .artifactAccount(artifact.getArtifactAccount())
+                    .customKind(artifact.isCustomKind())
+                    .location(artifact.getLocation())
+                    .metadata(artifact.getMetadata())
+                    .name(artifact.getName())
+                    .provenance(artifact.getProvenance())
+                    .type(artifact.getType())
+                    .version(artifact.getVersion())
+                    .build();
+              }).collect(Collectors.toList());
+        } catch (IOException e) {
+            log.error("Error setting references in artifacts  " + e.getMessage());
+        }
+        try {
+            for (Artifact ar : artifacts) {
+                downloadArtifactToTmpFileStructure(env, ar);
+            }
+        } catch (IOException e) {
+            throw new IllegalStateException("Failed to fetch kustomize files: " + e.getMessage(), e);
+        }
+
+        List<String> command = new ArrayList<>();
+            command.add("kustomize");
+            command.add("build");
+            command.add(templatePath);
+        result.setCommand(command);
+
+        return result;
+    }
+
+    private HashSet<String> getFilesFromArtifact(Artifact artifact) throws IOException {
+        Path artifactPath = Paths.get(artifact.getReference());
+        Kustomization kustomization = kustomizationFileReader.getKustomization(artifact);
+        filesToDownload.add(kustomization.getReference());
+        filesToDownload.addAll(kustomization.getFilesToDownload(artifactPath));
+        if (kustomization.getFilesToEvaluate() != null && !kustomization.getFilesToEvaluate().isEmpty()) {
+            for (String evaluate : kustomization.getFilesToEvaluate()) {
+                if (evaluate.contains(".")) {
+                    String tmp = evaluate.substring(evaluate.lastIndexOf(".") + 1);
+                    if (!tmp.contains("/")) {
+                        filesToDownload.add(artifactPath.resolve(evaluate).toString());
+                    } else {
+                        artifact.setReference(getSubFolder(evaluate, artifactPath));
+                        getFilesFromArtifact(artifact);
+                    }
+                } else {
+                    artifact.setReference(getSubFolder(evaluate, artifactPath));
+                    getFilesFromArtifact(artifact);
+                }
+            }
+        }
+        return filesToDownload;
+    }
+
+    private String getSubFolder(String pRelativePath, Path pPath) {
+        String basePath = pPath.toString();
```

Comment:
What does the method achieve? Is it just `pPath.resolve(pRelativePath).toString()`?
If not there must be a better way than replacing characters with hardcoded path separators.

Reply:
Lets say you want to resolve this "../../base" with this url "https:/api.github.com/repos/kubernetes-sigs/kustomize/contents/examples/ldap/overlays/production” the result should be “https:/api.github.com/repos/kubernetes-sigs/kustomize/contents/examples/ldap/base"

Reply:
never mind I found this method: `FilenameUtils.normalize`